src/App.tsx

```diff
@@ -58,11 +61,13 @@ type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
 type NumberMode = "normal" | "corner";
 
 interface BoardState {
-    highlights: Set<SquareAddress>;
+    continueHighlighting: boolean;
     contradicts: Set<SquareAddress>;
-    values: Map<SquareAddress, SquareValue>;
+    highlighting: boolean;
+    highlights: Set<SquareAddress>;
     markingMap: Map<SquareAddress, Set<SquareValue>>;
     numpadMode: NumberMode;
+    values: Map<SquareAddress, SquareValue>;
```

Comment:
These are getting too complicated to understand by name.

Please add a full jsdoc for each explaining why it exists.